charts/geoserver/v0.3.2/values.yaml

```diff
@@ -0,0 +1,132 @@
+image:
+  registry: docker.io
+  repository: kartoza/geoserver
+  tag: "2.17.0"
+  pullPolicy: IfNotPresent
+
+geoserverDataDir: /opt/geoserver/data_dir
+geowebcacheCacheDir: /opt/geoserver/data_dir/gwc
+
+geoserverUser: admin
+
+# For prefilled password
+# It will be stored in secret later
+geoserverPassword:
+
+# Use this if you have predefined secrets
+existingSecret: |
+
+# Define this for extra GeoServer environment variables
+extraPodEnv: |
+#  - name: KEY_1
+#    value: "VALUE_1"
+#  - name: KEY_2
+#    value: "VALUE_2"
+
+# This will be evaluated as pod spec
+extraPodSpec: |
+#  nodeSelector:
+#    a.label: value
+
+# Define this for extra secrets to be included
+extraSecret: |
+#  key_1: value_1
+
+# Define this for extra config map
+extraConfigMap: |
+# file_1: conf content
+
+# Define this for extra volume mounts in the pod
+extraVolumeMounts: |
+# You may potentially mount a config map/secret
+#  - name: custom-config
+#    mountPath: /docker-entrypoint.sh
+#    subPath: docker-entrypoint.sh
+#    readOnly: true
+
+# Define this for extra volume (in pair with extraVolumeMounts)
+extraVolume: |
+# You may potentially mount a config map/secret
+#  - name: custom-config
+#    configMap:
+#      name: geoserver-config
+
+service:
+  ## GeoServer service type
+  type: ClusterIP
+  # clusterIP: ""
+  # externalIPs: ""
+  port: 80
+
+  # nodePort:
+
+  labels: |
+  annotations: |
+
+  # loadBalancerIP:
+
+  # loadBalancerSourceRanges:
+  # - 10.10.10.0/24
+
+ingress:
+  enabled: false
+  annotations: {}
+  labels: {}
+  host: geoserver.hostname
+  tls:
+    enabled: false
+    ## Provide secret name if tls is enabled
+    secretName: geoserver-tls
+
+# probe can be overridden
+probe: |
+
+## You can also set global storage class
+# global.storageClass: "-"
+
+persistence:
+  geoserverDataDir:
+    enabled: true
+    existingClaim: false
+    mountPath: /opt/geoserver/data_dir
+    subPath: "data_dir"
+    size: 8Gi
+    # storageClass: "-"
+    accessModes:
+      - ReadWriteOnce
+    annotations: {}
+  geowebcacheCacheDir:
+    enabled: true
+    existingClaim: false
+    mountPath: /opt/geoserver/data_dir/gwc
+    subPath: "gwc"
+    size: 8Gi
```

Comment:
We are hardcoding the volume sizes i.e 8GI. Isn't there a way to define them to be generic i.e use env variable to enable a user to customise them ?

Reply:
These are default values and can be updated before deployment.